Source/WebCore/html/HTMLAttachmentElement.h

```diff
@@ -128,7 +129,12 @@ class HTMLAttachmentElement final : public HTMLElement {
     RefPtr<Image> m_icon;
     FloatSize m_iconSize;
 
-    RefPtr<HTMLAttachmentElement> m_innerLegacyAttachment;
+    // The thumbnail is shown if non-empty, otherwise the icon is shown if non-empty.
+    Vector<uint8_t> m_thumbnailImgSrcData;
+    LayoutSize m_requestedIconSize; // {w<0||h<0} = no need to request, {0,0} = need request, {w>0,w>0} = requesting.
```

Comment:
I don't think we need this three-state size. I'd just name this `m_iconSize`

Reply:
`m_iconSize` contains the actual size of the current icon (if any).
This `m_requestedIconSize` is what makes `requestWideLayoutIcon()` cheap to call multiple times, even while a request is in flight.

Reply:
Can we replace this with a boolean so we don't duplicate the icon size? Something like `m_needsIconRequest`?

Reply:
Ok, since the img size is fixed, I guess we don't need the "currently requesting a specific size" state.